src/main/java/org/broadinstitute/hellbender/tools/spark/pathseq/PSBwaUtils.java

```diff
@@ -43,6 +45,25 @@ static void addReferenceSequencesToHeader(final SAMFileHeader header,
     }
 
     /**
+     * Returns header with sequences that were aligned to at least once in reads
+     */
+    static SAMFileHeader removeUnmappedHeaderSequences(final SAMFileHeader header,
+                                                       final JavaRDD<GATKRead> reads,
+                                                       final Logger logger) {
+        final List<String> usedSequences = PSBwaUtils.getAlignedSequenceNames(reads);
+        usedSequences.forEach(seqName -> {
+            if (header.getSequence(seqName) == null)
+                logger.warn("One or more reads are aligned to sequence " + seqName + " but it is not in the header");
```

Comment:
you could probably fold this logger into the stream process below if you wanted, perhaps by making the `filter` call take a lambda that logs a warning if it returns false.